src/test/soundsourcemixin.h

```diff
@@ -0,0 +1,12 @@
+#pragma once
+#include "sources/soundsourceproxy.h"
+
+class SoundSourceMixin {
```

Comment:
How about naming it `SoundSourceProviderRegistration`? With a doc comment stating that this class is supposed to be used as a mixin for test classes that require accessing SoundSources.

Reply:
`SoundSourceProviderRegistrationMixin`?

Reply:
However. It does not need to be used strictly as a mixin. You could also instantiate the class a simple member.

Reply:
It's just the RAII pattern. You only need to use it as mixin if the constructor must run before another base class. But this depends on the usage context and is not an inherent restriction of the class as such.